phoenix-core/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -763,6 +770,18 @@ protected QueryPlan compileSingleFlatQuery(
         if(plan instanceof BaseQueryPlan){
             ((BaseQueryPlan) plan).setApplicable(isApplicable);
         }
+        if (context.getScan().getFilter() != null &&
```

Comment:
Maybe we can push this filter in preScannerOpen of the RegionScanner in Coproc, then it will be an all server-side fix

Reply:
added